services/pdf-ms/lambdas/addPdfToCase.ts

```diff
@@ -21,16 +21,18 @@ import log from '../../../libs/logs';
 
 const PDF_SSM_PARAMS = params.read(config.pdf.envsKeyName);
 
-// Convert DynamoDB item to plain object
-const dynamoDbConverter = AWS.DynamoDB.Converter;
+const dynamoDbItemToObjectConverter = AWS.DynamoDB.Converter;
 
 export async function main(
   event: Record<string, any>,
   context: Record<string, any>
 ): Promise<Boolean> {
-  const submittedCase: Case = dynamoDbConverter.unmarshall(event.detail.dynamodb.NewImage);
+  const submittedCase: Case = dynamoDbItemToObjectConverter.unmarshall(
+    event.detail.dynamodb.NewImage
```

Comment:
We should add safe check here so that contents in detail is not undefined.

```JS
function noSuchProperty(key, object) {
  // test if a property doesn't exist in the current object
  return !object.hasOwnProperty(key)
}

if (noSuchProperty('dynamodb', event.detail) && ){
  // do something
}

const {dynamodb} = event.detail

 if(noSuchProperty('NewImage', dynamodb)) {
   // do something
 }

```

Reply:
Will not fix. This service is prune for deletion :)

Reply:
Will be fixed in future task